lib/sequelize.js

```diff
@@ -1145,7 +1145,7 @@ module.exports = (function() {
             return transaction.commit().then(function () {
               resolve(result);
             });
-          }).catch(function (err) {
+          }).then(null, function (err) {
```

Comment:
Nitpicking sorry - but this could just be written as `then(function () commit, function rollback` instead of two separate thens.

Please change and squash into a single commit, then I'll be happy to merge